develop-docs/ingestion/symbolicator/index.mdx

```diff
@@ -0,0 +1,235 @@
+- `load(&self, data: ByteView<'static>) -> CacheEntry<Self::Item>` loads an instance of `Self::Item` from the given `data`.
+
+You can then add an instance of `Cacher<MyItemRequest>` to whichever service requires the cache and use its `compute_memoized` method to compute an instance of `Item`, automatically using the in-memory, on-disk, and shared caches.
+
+### `CacheError` / `CacheEntry`
+
+We only have one internal error type in Symbolicator: `CacheError`. This is its definition:
+
+```rust
+pub enum CacheError {
+    NotFound,
+    PermissionDenied(String),
+    Timeout(Duration),
+    DownloadError(String),
+    Malformed(String),
+    InternalError,
+}
+```
+
+Each of the cases represents something that can go wrong when fetching a file from somewhere. In addition, we have the type `pub type CacheEntry<T = ()> = Result<T, CacheError>` as an abbreviation.
+
+Once upon a time we had many individual error types for different parts of Symbolicator, but they were very hard to understand and we never really needed so many fine-grained distinctions, so we boiled them all down to common failure cases.
+
+### Expiry & Cleanup
+
+By default, we keep different kinds of cached files around for different lengths of time:
+
+- Downloaded and derived files are retained for 12 hours after their last use.
+- When a file can’t be downloaded, that fact is cached for 1 hour.
+- When a file is malformed (i.e. can’t be processed for some reason), that fact is cached for 12 hours.
+
+Cache cleanup is performed by the `symbolicator cleanup` command. We run this command periodically as part of Symbolicator’s kubernetes deployment.
+
+### Versioning
+
+All the caches are also versioned. Each cache has both a current version number and a list of compatible versions. The current version is always tried first, then the compatible versions in order. The idea behind this is that we may make improvements to a data format that still leave the previous version readable. If there is only an older (but still compatible) version available, we use it, but also spawn a computation to update it to the current version. If there are only incompatible versions available, that counts as not found.
+
+## Deployment
+
+In SaaS, we deploy 9 groups/environments of Symbolicator in total: for each platform (`native` , `js`, `jvm`) we have production, canary, and LPQ Symbolicators.
+
+We have separate deployments per platform so that we can scale them independently and e.g. the huge volume of JS requests doesn’t starve native requests. All platforms use the same build of Symbolicator; in principle, each of them could service any kind of request.
+
+### The Low Priority Queue
+
+A project can be assigned to the *low priority queue* for a platform if it uses an unusual amount of symbolication time. We have Symbolicator pods set aside for projects on the LPQ so that such projects don’t block symbolication for other projects.
```

Comment:
This paragraph is no longer relevant; the LPQ has been removed since this document was written.
```suggestion
```